### ISSUE TITLE
real world application idea

### ISSUE THREAD

**Issue by markz-nyc** [yusuketomoto/chainer-fast-neuralstyle#45]
Hi guys, just start to play this fast neural style recently, it's amazing to get it work on my laptop with geforce 940M 2G GPU, although it's slow. I am thinking to turn this algorithm into real word application. Besides app like prisma, it seem we can easily create animation movie in the future with such technology. Any other thought about building real world product?

BTW, have fun with modeling and generating.

**Comment by rogozin70** [comment]
I started the server.
http://photofuneditor.com/prisma-dreamscope-filters
Add your filters-models to the forum and I'll include them in the online processing.

**Comment by markz-nyc** [comment]
really cool, I wonder how you build the server, by using AWS GPU?

**Comment by rogozin70** [comment]
GPU- gtx 970 - 4 gb.

**Comment by GitResumeCoding** [comment]
I just tested it, that's very fast! Well done. I'd love to have some more setup details, over been considering that gpu card, what are some tweaks you did and pitfalls ran into. Or is that seriously an out of the box result?

**Comment by markz-nyc** [comment]
Yeah, I tried it and it's fast. Very impressive that you only use gtx970 4g. Did you setup a home server, would like to know more detail about your backend setup.

**Comment by 6o6o** [comment]
@markz-nyc So what's the application? Transforming videos is nothing new and has been done previously numerous times. Here's my recent attempt, which includes optical flow for frame blending https://youtu.be/h0jH0bJIvcM

**Comment by rogozin70** [comment]
I have installed CUDA and CUDNN. Everything is done according to the instructions.
@6o6o I noticed in your video new filters. You do not have share them?

**Comment by 6o6o** [comment]
Yea, I can't, sorry. I don't own them, they were trained by other people and are used in an app.

**Comment by markz-nyc** [comment]
@rogozin70, I wonder how to turn that into that web application, I mean user upload a photo in the web and then get the stylized photo output, do you mind share more detail about setup such application, for example, did you rent a server or use your own machine to do the process?

**Comment by rogozin70** [comment]
I think to buy your own home server will be more profitable. 
Prices are too high on Amazon.

**Comment by nandexsp** [comment]
hello @markz-nyc , I think like @rogozin70 , the server via AWS is too expensive and you could by your own server and compute inside.

**Comment by markz-nyc** [comment]
@nandexsp, Yeah I am currently working on setting up my own server, I have 0 server side knowledge, so I am doing research about framework like django and flask now.

**Comment by DylanAlloy** [comment]
@markz-nyc I made exactly what you're talking about in my repository.

https://github.com/DylanAlloy/NeuralStyle-WebApp

**Comment by rogozin70** [comment]
aws 1$ - 3 hours. 1 day - 8$, 1 month - 240 $.  Cost home server (icore3 +gtx 970) = 800$
3 months = 1 server

**Comment by DylanAlloy** [comment]
@rogozin70 Completely unscalable, runs off of your own internet connection which will bottleneck everything else, will eventually destroy your router unless you buy a good one, you might as well buy business internet, etc.

If you're looking to host a website that lots of people are going to use to run an algorithm that can take up to 30GB of RAM, you're going to need more than a home server.

**Comment by markz-nyc** [comment]
right now, it seems take me 27 hours to train a model, my room temp is too high, seems I have to move the machine into the fridge, lol

**Comment by markz-nyc** [comment]
@DylanAlloy  nice repo, I will take a look when I get a chance. Yeah, AWS is a better choice considering it as real product rather than hobby.

**Comment by nandexsp** [comment]
You have a niche repo @DylanAlloy , are you training with epoch 2, or higher??

**Comment by DylanAlloy** [comment]
@nandexsp @markz-nyc thanks, yeah I am using 2 epochs. Right now it's taking 73GB of RAM to generate a 4K image. I'm hoping it doesn't go over 122GB, that's all I have for now...

**Comment by nandexsp** [comment]
Great Work man!

**Comment by markz-nyc** [comment]
Prisma already made the neural style into offline mode, how iPhone can get such good result in few seconds without using gpu?

**Comment by DylanAlloy** [comment]
Because it's based on C code probably.

**Comment by xpeng** [comment]
@DylanAlloy wow, great work! you generated a such large size image! does the style or style scale been applied looks as similar with small size image for example 720pixel?

**Comment by 6o6o** [comment]
@DylanAlloy I think it's good-old python. The key action here is downsampling. Original implementation already does this twice using strided convolutions. So, when you process a 512x512 image, the actual transformation is applied to 128x128 image. I guess Prisma took this a bit further by increasing stride or adding another pair of convolution / deconvolution layers, which downsample more, saving on memory consumption, increasing computation speed and lastly, enlarging style features.

**Comment by markz-nyc** [comment]
@6o6o Do you want to do this as freelance job by helping me setup those up?

**Comment by 6o6o** [comment]
Thanks, I'm already busy trying various experiments on anything i can think of. I'm not really an expert in the field, so it's largely trial and error way and most of them end up futile. Nevertheless, I'm super excited about the technology and will keep you guys informed on any success stories, if any occur :)

**Comment by rupeshs** [comment]
Check out my NeuralStyler app 
Turn Your Videos/Photos/GIF into Art
http://neuralstyler.com/
https://github.com/rupeshs/neuralstyler

**Comment by nandexsp** [comment]
Its a good app @rupeshs , you could checks apps like prisma or pixify. The last one has video in iOS and more customize options.
